orttraining/orttraining/test/training_api/common/synthetic_data_loader.h

```diff
@@ -74,24 +57,29 @@ struct SyntheticSampleBatch {
   void AddInt32Input(const std::vector<int64_t>& shape, int32_t low, int32_t high);
   void AddInt64Input(const std::vector<int64_t>& shape, int64_t low, int64_t high);
   void AddFloatInput(const std::vector<int64_t>& shape);
+  void AddBoolInput(const std::vector<int64_t>& shape);
 
   size_t NumOfInput() {
-    return data_vector_.size();
+    return input_count_;
   }
 
-  SyntheticInput* GetInputAtIndex(size_t index) {
-    return data_vector_[index].get();
+  SyntheticInput& GetInputAtIndex(size_t index) {
+    return data_vector_[index];
   }
```

Comment:
There are multiple issues with this function.
1) operator[] does not perform range checks and neither is the code.
2) There are specific interfaces to add data to this class that properly increment input_count_ and may perform other checks, however, with this method anyone can get a mutable reference and do whatever they please with it.

Reply:
You are right! I will make the return value being const ref. Thanks!

Reply:
We need to pass non const pointer of the data to C API. I moved the logic SyntheticSampleBatch--> Batch of OrtValue into SyntheticSampleBatch::GetBatch, then those APIs including GetInputAtIndex() are not needed.